Find a functionally equivalent code:android.widget.RelativeLayout.LayoutParams.resolveRules:COMMENT
Method Modifier: private     
Comment:// only the "left"/"right" rules at the end.

@@ -44,50 +44,50 @@
             }
             mRules[ALIGN_PARENT_START] = 0;
         }
-        if (mRules[ALIGN_PARENT_RIGHT] == 0) {
+        if (mRules[ALIGN_PARENT_END] != 0) {
             if (mRules[ALIGN_PARENT_RIGHT] == 0) {
                 // "right" rule is not defined but "end" rule is: use the "end" rule as the
                 // "right" rule

